### PR TITLE
Issue #240: add breakdown of pom. Resolves #240

### LINKED ISSUE #240: Create Guide: Breaking Down the Boosters
Explain, in the context of an example Booster, what is being set up for a user (so they might manually create a project on their own)

* pom.xml elements
* Relevant application code
* fabric8-maven-plugin config

### PR DESCRIPTION
**Pull request by rhoads-zach**
@inoxx03 mind taking  quick look?

### CODE REVIEW

File: docs/topics/http-api-vertx-pom.adoc [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+          </plugin>
+        </plugins>
+      </build>
+    </profile>
+  </profiles>
+</project>
+----
+
+<1> link:https://github.com/openshiftio/booster-parent/blob/master/pom.xml[Parent pom] which contains the necessary dependencies and the required versions for the booster projects.
+<2> These are properties for setting the {VertX} and {VertX} maven plugin version to use as well as the name of the {VertX} vertical.
```
Comment (Ladicek) [inline docs/topics/http-api-vertx-pom.adoc]: Not "vertical" but "verticle".

File: docs/topics/http-api-vertx-pom.adoc [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+<5> link:https://github.com/vert-x3/vertx-web/blob/master/pom.xml[Pom] containing the components for building web applications with {VertX}.
+<6> Various dependencies needed for testing the application. This includes several {VertX} components such as link:http://vertx.io/docs/vertx-web-client/java/[vertx-web-client] and link:http://vertx.io/docs/vertx-unit/java/[vertx-unit] as well as other projects such as  link:https://joel-costigliola.github.io/assertj/[assertj] for assertions, link:https://github.com/rest-assured/rest-assured[rest-assured] for testing REST services, and link:https://github.com/awaitility/awaitility[awaitility] for doing asynchronous operations.
+<7> link:https://vmp.fabric8.io[{VertX} Fabric8 Maven Plugin] used for packaging and deploying {VertX} applications
+<8> link:https://wiki.eclipse.org/M2E_compatible_maven_plugins[Lifecycle metadata] that works with the {VertX} Fabric8 Maven Plugin for building and deploy the application.<9> This is the profile for building and deploying the booster to OpenShift. It uses the link:http://fabric8.io/gitbook/mavenPlugin.html[Fabric8 Maven Plugin (fmp)] to build and deploy the application with the S2I Build Process.
```
Comment (Ladicek) [inline docs/topics/http-api-vertx-pom.adoc]: This line contains `<9>` which is then repeated below on an extra line.

File: docs/topics/http-api-wf-swarm-pom.adoc [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+        </plugins>
+      </build>
+    </profile>
+  </profiles>
+</project>
+----
+
+<1> link:https://github.com/openshiftio/booster-parent/blob/master/pom.xml[Parent pom] which contains the necessary dependencies and the required versions for the booster projects.
+<2> These are properties for setting the {WildFlySwarm} and RESTEasy version.
+<3> link:https://github.com/wildfly-swarm/wildfly-swarm/blob/master/boms/bom/pom.xml[Parent Bom] for building {WildFlySwarm} applications with maven.
```
Comment (Ladicek) [inline docs/topics/http-api-wf-swarm-pom.adoc]: Not a "parent BOM", just a "BOM".

File: docs/topics/http-api-wf-swarm-pom.adoc [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+<4> link:https://jcp.org/en/jsr/detail?id=339[JAX-RS] dependency for creating REST services.
```
Comment (Ladicek) [inline docs/topics/http-api-wf-swarm-pom.adoc]: It's actually a Swarm fraction. One that brings JAX-RS support.

File: docs/topics/http-api-wf-swarm-pom.adoc [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+<5> link:https://docs.jboss.org/resteasy/docs/3.1.3.Final/userguide/html_single/#RESTEasy_Client_Framework[RESTEasy client] dependency for connecting to other REST services which is used in testing.
+<6> link:http://arquillian.org[Arquillian] dependency for testing.
```
Comment (Ladicek) [inline docs/topics/http-api-wf-swarm-pom.adoc]: Again this is actually a Swarm fraction. One that brings an Arquillian adapter for Swarm.

File: docs/topics/http-api-wf-swarm-pom.adoc [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+<7> link:http://arquillian.org/guides/getting_started/#add_the_arquillian_apis[Arquillian APIs] for testing.
```
Comment (Ladicek) [inline docs/topics/http-api-wf-swarm-pom.adoc]: I'd add "Arquillian APIs for testing with JUnit".

**Comment by Ladicek** [comment]
I've also seen a couple of capitalization issues. Things like "maven" (should be Maven), "tomcat" (should be Tomcat or "Openshift" (should be OpenShift).

**Comment by Ladicek** [comment]
CC @SlavoKrupa @RadekKoubsky

File: docs/topics/http-api-spring-boot-pom.adoc [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+<6> link:https://github.com/spring-projects/spring-boot/tree/v1.4.1.RELEASE/spring-boot-starters/spring-boot-starter-actuator/pom.xml[Starter pom] for using {SpringBoot}’s actuator. The actuator provides monitoring and management for your application.
+<7> link:https://github.com/spring-projects/spring-boot/tree/v1.4.1.RELEASE/spring-boot-starters/spring-boot-starter-test/pom.xml[Starter pom] for testing Spring Boot applications with libraries including JUnit, Hamcrest and Mockito. 
+<8> This maven goal is part of the {SpringBoot} maven plugin and will repackage the existing jar file that is created during the `package` goal.
+<9> This is the profile for building and deploying the booster to OpenShift. It uses the link:http://fabric8.io/gitbook/mavenPlugin.html[Fabric8 Maven Plugin (fmp)] to build and deploy the application with the S2I Build Process.
+<10> This profile is for running integration tests when the application is running on Openshift. For example, a test may require a database pod or a ConfigMap value.
```
Comment (inoxx03) [inline docs/topics/http-api-spring-boot-pom.adoc]: _Openshift_ > _OpenShift_

File: docs/topics/http-api-spring-boot-pom.adoc [hunk 8]
```diff
@@ -0,0 +1,174 @@
+[[breakdown_pom]]
+= Breakdown of POM components
```
Comment (inoxx03) [inline docs/topics/http-api-spring-boot-pom.adoc]: This h1 should probably be in title case.

File: docs/topics/http-api-vertx-pom.adoc [hunk 9]
```diff
@@ -0,0 +1,208 @@
+[[breakdown_pom]]
+= Breakdown of POM components
```
Comment (inoxx03) [inline docs/topics/http-api-vertx-pom.adoc]: Title case?

File: docs/topics/http-api-wf-swarm-pom.adoc [hunk 10]
```diff
@@ -0,0 +1,172 @@
+[[breakdown_pom]]
+= Breakdown of POM components
```
Comment (inoxx03) [inline docs/topics/http-api-wf-swarm-pom.adoc]: Title case?

File: docs/topics/http-api-spring-boot-pom.adoc (hunk 7 above)
Comment (inoxx03) [inline docs/topics/http-api-spring-boot-pom.adoc]: OpenShift should be camel case.

File: docs/topics/http-api-vertx-pom.adoc (hunk 1 above)
Comment (inoxx03) [inline docs/topics/http-api-vertx-pom.adoc]: maven > Maven

File: docs/topics/http-api-wf-swarm-pom.adoc [hunk 11]
```diff
@@ -0,0 +1,172 @@
...
+            </executions>
+          </plugin>
+        </plugins>
+      </build>
+    </profile>
+  </profiles>
+</project>
+----
+
+<1> link:https://github.com/openshiftio/booster-parent/blob/master/pom.xml[Parent pom] which contains the necessary dependencies and the required versions for the booster projects.
```
Comment (inoxx03) [inline docs/topics/http-api-wf-swarm-pom.adoc]: should pom be stylized in all caps as POM?

File: docs/topics/http-api-wf-swarm-pom.adoc [hunk 12]
```diff
@@ -0,0 +1,172 @@
...
+<4> link:https://jcp.org/en/jsr/detail?id=339[JAX-RS] dependency for creating REST services.
+<5> link:https://docs.jboss.org/resteasy/docs/3.1.3.Final/userguide/html_single/#RESTEasy_Client_Framework[RESTEasy client] dependency for connecting to other REST services which is used in testing.
+<6> link:http://arquillian.org[Arquillian] dependency for testing.
+<7> link:http://arquillian.org/guides/getting_started/#add_the_arquillian_apis[Arquillian APIs] for testing. 
+<8> link:https://wildfly-swarm.gitbooks.io/wildfly-swarm-users-guide/content/v/2017.6.0/getting-started/tooling/maven-plugin.html[{WildFlySwarm} Maven Plugin] used to create the application link:https://wildfly-swarm.gitbooks.io/wildfly-swarm-users-guide/content/v/2017.6.0/getting-started/concepts.html[uberjar].
+<9> This is the profile for building and deploying the booster to OpenShift. It uses the link:http://fabric8.io/gitbook/mavenPlugin.html[Fabric8 Maven Plugin (fmp)] to build and deploy the application with the S2I Build Process.
+<10> This profile is for running integration tests when the application is running on Openshift. For example, a test may require a database pod or a ConfigMap value.
```
Comment (inoxx03) [inline docs/topics/http-api-wf-swarm-pom.adoc]: Openshift > OpenShift

File: docs/topics/http-api-wf-swarm-pom.adoc (hunk 3 above)
Comment (inoxx03) [inline docs/topics/http-api-wf-swarm-pom.adoc]: Also,
maven > Maven

**Comment by rhoads-zach** [comment]
Thank you for all the feedback @Ladicek, I've amended this PR with the changes.

Let me know what you think.

File: docs/topics/http-api-spring-boot-pom.adoc [hunk 13]
```diff
@@ -0,0 +1,174 @@
...
+    </profile>
+  </profiles>
+</project>
+----
+
+//http://docs.spring.io/spring-boot/docs/1.4.1.RELEASE/reference/htmlsingle/#using-boot-starter
+
+<1> link:https://github.com/openshiftio/booster-parent/blob/master/pom.xml[Parent pom] which contains the necessary dependencies and the required versions for the booster projects.
+<2> These are properties for setting the {SpringBoot} and Tomcat version to use.
+<3> link:https://github.com/spring-projects/spring-boot/blob/v1.4.1.RELEASE/spring-boot-parent/pom.xml[Parent pom] for building {SpringBoot} applications with Maven. More details on using the parent pom are available in the link:http://docs.spring.io/spring-boot/docs/1.4.1.RELEASE/reference/htmlsingle/#using-boot-maven-parent-pom[{SpringBoot} documentation].
```
Comment (Ladicek) [inline docs/topics/http-api-spring-boot-pom.adoc]: Actually this is not a parent POM, it's a BOM.

File: docs/topics/http-api-vertx-pom.adoc [hunk 14]
```diff
@@ -0,0 +1,208 @@
...
+        </plugins>
+      </build>
+    </profile>
+  </profiles>
+</project>
+----
+
+<1> link:https://github.com/openshiftio/booster-parent/blob/master/pom.xml[Parent pom] which contains the necessary dependencies and the required versions for the booster projects.
+<2> These are properties for setting the {VertX} and {VertX} Maven plugin version to use as well as the name of the {VertX} verticle.
+<3> link:https://github.com/vert-x3/vertx-dependencies/blob/master/pom.xml[Pom] for building {VertX} applications with Maven.
```
Comment (Ladicek) [inline docs/topics/http-api-vertx-pom.adoc]: This is also a BOM.

**Comment by Ladicek** [comment]
I'm not sure about capitalization of "pom" (maybe should be POM?) and fmp (maybe should be FMP?). Otherwise LGTM.

**Comment by inoxx03** [comment]
merging

**Comment by inoxx03** [comment]
merged: 37d38711933d74820d54c7b2b8a427a09653859f

**Pull request closed without merging**